core/web/evm_transfer_controller.go

```diff
@@ -104,11 +105,14 @@ func ValidateEthBalanceForTransfer(c *gin.Context, chain evm.Chain, fromAddr com
 		return errors.Wrap(err, "failed to estimate gas")
 	}
 
-	// TODO: support EIP-1559 transactions
-	if fees.Legacy == nil {
-		return errors.New("estimator did not return legacy tx fee estimates")
+	var gasPrice *assets.Wei
+	if fees.Legacy != nil {
+		gasPrice = fees.Legacy
+	} else if fees.Legacy == nil && fees.DynamicFeeCap != nil && fees.DynamicTipCap != nil {
```

Comment:
Can drop the `== nil`, no?
```suggestion
	} else if fees.DynamicFeeCap != nil && fees.DynamicTipCap != nil {
```

Reply:
ah yeah - I added that as a sanity check, but it can probably be dropped